File: backend/src/main/java/net/dnadas/training_portal/dto/group/project/questionnaire/SubmittedQuestionResponseDto.java
package net.dnadas.training_portal.dto.group.project.questionnaire;

import net.dnadas.training_portal.model.group.project.questionnaire.QuestionType;

import java.util.List;

public record SubmittedQuestionResponseDto(
  Long id, String text, QuestionType type, Integer receivedPoints, Integer maxPoints, Integer order,
  List<SubmittedAnswerResponseDto> answers) {
}

File: frontend/src/questionnaires/dto/SubmittedQuestionResponseDto.ts
import {QuestionType} from "./QuestionType.ts";
import {SubmittedAnswerResponseDto} from "./SubmittedAnswerResponseDto.ts";

export interface SubmittedQuestionResponseDto {
  id: number;
  text: string;
  type: QuestionType;
  receivedPoints: number;
  maxPoints: number;
  order: number;
  answers: SubmittedAnswerResponseDto[];
}
